js/ui/tab_cookies.js

```diff
@@ -371,19 +423,89 @@
     };
   }
 
-  function createDomainObj(domain, cookienum)
+  function createDomainObj(domain, cookienum, whitelist = false)
   {
     return {
       dataset: {
-        access: domain
+        access: domain,
+        whitelist: whitelist
       },
       texts: {
         domain: domain,
         cookienum: cookienum + " Cookies"
       }
     };
   }
-
+  function updateWhitelistInList(whitelistChange)
+  {
+    for (var domain in whitelistChange)
+    {
+      let domainElement = tableList.getItem(domain);
+      if (domainElement) 
+      {
+        domainElement.dataset.whitelist = whitelistChange[domain].domainWhitelist;
+        tableList.updateItem(domainElement, domain);
+        let cookies = whitelistChange[domain].cookies;
+        if (domainElement.subItems) 
+        {
+          for (let subElement of domainElement.subItems) 
+          {
+            if (cookies.includes(subElement.texts.name)) 
+            {
+              subElement.dataset.whitelist = true;
+              let cookieAccessor = "li[data-access='" + subElement.dataset.access + "']"
+              document.querySelector(cookieAccessor).dataset.whitelist = true
+            } 
+            else 
+            {
+              subElement.dataset.whitelist = false;
+              let cookieAccessor = "li[data-access='" + subElement.dataset.access + "']"
+              document.querySelector(cookieAccessor).dataset.whitelist = false
+            }
+          }
+        }
+      }
+    }
+  };
+  onStorageChange.addListener(function(changeInfo)
+  {
+    if ("cookieWhitelist" in changeInfo)
+    {
+      let newValue = changeInfo.cookieWhitelist.newValue
+      let oldValue = changeInfo.cookieWhitelist.oldValue
+      let changed = {}
+      if (oldValue) 
+      {
+        for (let key in newValue)
```

Comment:
I wonder if something like this would be enough:

```
const diff = Object.keys(newValue).reduce((acc, domain) => 
  {
    if (JSON.stringify(oldValue[domain]) != JSON.stringify(newValue[domain]))
    {
      acc[domain] = newValue[domain];
    }
    return acc;
  }, {});
```

Considering that out JSON objects are not big and complex, I don't think that Stringify can come too much costy, but if you would have that concern you can use `reduce` and simplify the existing logic, what you think ?

If needed can provide example for current implementation simplification.

Reply:
this definitely seems like an improvement
didn't know about the existence reduce, but seems like a better way
JSON.stringify can definitely reduce the complexity of the code a lot
the only downside i see is that if somehow the order of the cookies in oldValue[domain].cookies changes it would consider it a changed object, while the current implementation would consider them the same, don't think that would have any real impact tho
it would definitely be an improvement overall, will check it out and change it later

Reply:
You are right, even though in most of the cases it would work, but mostly because how v8 might deal with the simple Objects and order of the keys in the object, but that should work more reliable for the Arrays as we only have String values, so maybe something like this can be even better solution:

```
const newDomainObj = newValue[domain];
const oldDomainObj = oldValue[domain];

if (newDomainObj. domainWhitelist != oldDomainObj.domainWhitelist ||
    JSON.stringify(newDomainObj.cookies) != JSON. stringify(oldDomainObj.cookies))
{
 ... 
}
```